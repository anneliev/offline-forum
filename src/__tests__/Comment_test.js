import React from 'react';
import { shallow, mount } from 'enzyme';
import Comments from '../components/Comments';
import* as api from '../api';

const comments = [
  {
    comment: "A comment",
    id: "1",
    postId: 1,
    author: "Zac",
    date: "2018-01-23",
    currentPersona: "Zac"
  }
];

it('should show comment list', () => {
  const wrapper = mount(<Comments postId="1" currentPersona="Zac" author="Zac" />);
  wrapper.setState({ comments });
  wrapper.instance().renderCommentList(comments);
  expect(wrapper.find('SingleComment').text()).toContain("A comment");
})

it('should call remove comment function', () => {
  api.removeComment = jest.fn();
  const wrapper = shallow(<Comments postId="1" currentPersona="Zac" author="Zac"/>);
  wrapper.instance().removeComment();
  expect(api.removeComment).toHaveBeenCalled();
})