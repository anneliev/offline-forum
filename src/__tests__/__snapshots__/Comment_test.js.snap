// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<Comments
  currentPersona="Zac"
  postId="1"
>
  <div
    className="py-2"
  >
    <h2
      className="text-indigo-darker border-b mb-4"
    >
      Comments
    </h2>
    <CreateNewComment
      author="Zac"
      postId="1"
      updateComments={[Function]}
    >
      <form
        className="container mx-auto flex flex-col p-6"
        onSubmit={[Function]}
      >
        <label
          className="mb-2"
          htmlFor="comment"
        >
          Comment
        </label>
        <textarea
          className="shadow rounded border mb-2 p-2"
          id="comment"
          name="comment"
          onChange={[Function]}
          value=""
        />
        <input
          className="bg-indigo-dark hover:bg-indigo-darker text-white font-bold py-2 px-4 rounded-full"
          type="submit"
          value="Comment"
        />
      </form>
    </CreateNewComment>
  </div>
</Comments>
`;
